### PR TITLE
Update MuSE to latest version 2.1.2

### PR DESCRIPTION
**Pull request by famosab**
Updated the recipe for MuSE to the latest version 2.1 and changed the repository to its new location: https://github.com/wwylab/MuSE

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/muse/meta.yaml [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 package:
-  name: muse
-  version: 1.0.rc
+  name: {{ name|lower }}
+  version: {{ version }}
 
 source:
-  md5: c63fdb48c041f6f9545879f1a7e4da58
-  url: https://github.com/danielfan/MuSE/archive/v1.0-rc.tar.gz
+  sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```
Comment (mencian) [inline recipes/muse/meta.yaml:9]:
```suggestion
  sha256: 0767f6614a6f26146beffa3546eb7c1a7c2f87bd21d6001ac2e5d157182fe14c
```

**Comment by famosab** [comment]
@mencian did you manage to build this locally? Are you planning to continue this? Or do you have an idea why the build is still failing?

File: recipes/muse/build.sh [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
-cp MuSE "${PREFIX}/bin/"
+
+export INCLUDES="-I{PREFIX}/include"
+export LIBPATH="-L${PREFIX}/lib"
+export LDFLAGS="${LDFLAGS} -L${PREFIX}/lib"
+
+export CFLAGS="${CFLAGS} -O3"
+
+CC="${CC} -O3 ${LDFLAGS}" CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS} -O3" \
+	CPP="${CPP}" CPPFLAGS="${CPPFLAGS} -I${PREFIX}/include" \
```
Comment (martin-g) [inline recipes/muse/build.sh:12]:
```suggestion
	CPP="${CXX}" CPPFLAGS="${CXXFLAGS} -I${PREFIX}/include" \
```
By convention `CPP` is C pre-processor. 
For C++ compiler the usual name is `CXX`

File: recipes/muse/meta.yaml [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
-  md5: c63fdb48c041f6f9545879f1a7e4da58
-  url: https://github.com/danielfan/MuSE/archive/v1.0-rc.tar.gz
-  patches:
-    - 0001-add-LDFLAGS.patch
+  url: https://github.com/wwylab/MuSE/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
 
 build:
-  number: 8
+  number: 0
```
Comment (martin-g) [inline recipes/muse/meta.yaml:14]:
```suggestion
  number: 1
```

**Comment by martin-g** [comment]
> @mencian did you manage to build this locally? Are you planning to continue this? Or do you have an idea why the build is still failing?

`install-muse.sh` needs more love.
```
$SRC_DIR/boost_1_70_0/boost-build.jam:17: in module scope from module[0m
2024-06-10T11:12:22.9881805Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) [0m
2024-06-10T11:12:22.9882170Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: No toolsets are configured.[0m
2024-06-10T11:12:22.9882533Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: Configuring default toolset "gcc".[0m
2024-06-10T11:12:22.9885540Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: If the default is wrong, your build may not work correctly.[0m
2024-06-10T11:12:22.9885991Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: Use the "toolset=xxxxx" option to override our guess.[0m
2024-06-10T11:12:22.9886443Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: For more configuration options, please consult[0m
2024-06-10T11:12:22.9886921Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) warning: http://boost.org/boost-build2/doc/html/bbv2/advanced/configuration.html[0m
2024-06-10T11:12:22.9887449Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) $SRC_DIR/boost_1_70_0/tools/build/src/tools/gcc.jam:230: in gcc.init from module gcc[0m
2024-06-10T11:12:22.9887863Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) error: toolset gcc initialization:[0m
2024-06-10T11:12:22.9888295Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) error: no command provided, default command 'g++' not found[0m
2024-06-10T11:12:22.9888661Z 11:12:22 [32mBIOCONDA INFO[0m (OUT) error: initialized from[0m
```

It builds boost, gperftools and htslib without passing `${CC}`/`${CXX}`.
IMO the recipe should not use `install-muse.sh` but should add these as build/host dependencies and build only MUSE.

**Comment by famosab** [comment]
@jiyunmaths Maybe you can help here as I am not that familiar with cpp.

@maxulysse Pinging because you might know who to ask.

**Comment by famosab** [comment]
@martin-g How do we not use the install-muse.sh file when we use the tar from the release?

**Comment by martin-g** [comment]
> @martin-g How do we not use the install-muse.sh file when we use the tar from the release?

You will need to add `requirements > build` entries for boost, gperftools and htslib (and pin their versions if needed). Then in `recipe/muse/build.sh` do only the last steps from install_muse.sh, i.e. only build Muse assuming the dependencies are provided.

**Comment by famosab** [comment]
@coderabbitai review

**Comment by famosab** [comment]
We are still failing here:
```
07:58:25 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmplzupm39j/build_script.bash:/opt/build_script.bash -v /opt/mambaforge/envs/bioconda/conda-bld:/opt/host-conda-bld -v /home/vsts/work/1/s/recipes/muse:/opt/recipe -e LANG=C.UTF-8 -e HOST_USER_ID=1001 quay.io/bioconda/bioconda-utils-build-env-cos7:3.3.1 /bin/bash /opt/build_script.bash

07:58:25 BIOCONDA ERROR BUILD FAILED recipes/muse
```

**Comment by famosab** [comment]
@martin-g I had some more tries at this recipe and concluded with a collegue that somehow not all dependencies get loaded when needed. For example we found that `tabix` is missing. I tried running `bioconda-utils` with flags that enable keeping temporary files to inspect the build directories but unfortunately I was not able to locate it. Maybe you know how to debug that? I think we just need to have a correct declaration in this line and then it should work:
`export INCLUDES="-I${BUILD_PREFIX}/include -I${PREFIX}/include -I${SRC_DIR}/inc"`

**Comment by mencian** [comment]
Finally able to build `MuSE`! According to the docs, `MuSE` is only available on linux, so I've disabled the build on osx.

**Comment by famosab** [comment]
@mencian Thanks you so much 🥳

**Pull request merged**